src/recipe.h

```diff
@@ -50,6 +51,26 @@ struct recipe_proficiency {
     void deserialize( JsonIn &jsin );
 };
 
+struct book_recipe_data {
+    int skill_req = -1;
+    bool use_alt_name = false;
+    std::string alt_name = "";
+    bool hidden = false;
+
+    book_recipe_data( JsonArray arr ) {
+        int arr_size = arr.size();
+        if( arr_size > 1 ) {
+            skill_req = arr.get_int( 1 );
+        }
+        if( arr_size > 2 ) {
+            std::string name = arr.get_string( 2 );
+            alt_name = name;
+            use_alt_name = true;
+            hidden = name.empty();
```

Comment:
Please use a specialized json value for hiding instead of checking the emptiness of the name.

Reply:
Oof, while I do agree, that would likely mean switching book_learn from arrays to objects and thus probably adjusting all the recipes. I was hoping to get away with just reimplementing what's in the documentation.

Reply:
You can implement a new format while maintaining back-compatibility for the old format.

Reply:
After looking into it a bit more, this seems to be the only occurence of an array with individual optional parameters. It probably really should be replaced by an object, with backwards compatibility where you don't want to use alternate names or hide it.

Reply:
Updated it like that. Old format won't support alternate names or hidden recipes, but still works as before. Only updated one json file as an example.